sources/osgShader/node/lightCommon.glsl

```diff
@@ -0,0 +1,77 @@
+float getLightAttenuation(const in float dist, const in vec4 lightAttenuation) {
+    // lightAttenuation(constantEnabled, linearEnabled, quadraticEnabled)
+    // TODO find a vector alu instead of 4 scalar
+    float constant = lightAttenuation.x;
+    float linear = lightAttenuation.y * dist;
+    float quadratic = lightAttenuation.z * dist * dist;
+    return 1.0 / (constant + linear + quadratic);
+}
+
+#pragma DECLARE_FUNCTION
+void precomputeSpot(
+        const in vec3 normal,
+        const in vec3 viewVertex,
+                          
+        const in vec3 lightViewDirection,
+        const in vec4 lightAttenuation,
+        const in vec3 lightViewPosition,
+        const in float lightSpotCutOff,
+        const in float lightSpotBlend,
+                          
+        out float attenuation,
+        out vec3 eyeLightDir,
+        out float dotNL) {
+
+    attenuation = 0.0;
+    if (lightSpotCutOff <= 0.0) return;
+
+    eyeLightDir = lightViewPosition - viewVertex;
+    float dist = length(eyeLightDir);
+
+    eyeLightDir = dist > 0.0 ? eyeLightDir / dist : vec3( 0.0, 1.0, 0.0 );
+    float cosCurAngle = dot(-eyeLightDir, lightViewDirection);
+    if (cosCurAngle < lightSpotCutOff) return;
+
+    float spot = 1.0;
+    if (lightSpotBlend > 0.0) {
```

Comment:
perhaps we could take that refactor occasion to prevent in js code that lightSpotBlend is never near 0 and get rid